File: 1089-duplicate-zeros/1089-duplicate-zeros.py
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        n=len(arr)
        temp=[0]*n
        j=0
        for i in range(n):
            if j>=n:
                break
            temp[j]=arr[i]
            j=j+1
            if arr[i] ==0 and j<n:
                temp[j]=0
                j=j+1
        for i in range(n):
            arr[i]=temp[i]

        

File: 1089-duplicate-zeros/1089-duplicate-zeros.cpp
class Solution {
public:
    void duplicateZeros(vector<int>& arr) 
    {
        int n = arr.size();
        int j=0;
        vector<int>temp(n);
        for (int i=0;i<n && j<n;i++)
        {
            temp[j++]=arr[i];
            if (arr[i]==0 && j<n)
            {
                temp[j++]=0;
            }
        }
        arr =temp;
    }
};